diff -Naur libfreenect2-0.2.0-old/src/frame_listener_impl.cpp libfreenect2-0.2.0-new/src/frame_listener_impl.cpp
--- libfreenect2-0.2.0-old/src/frame_listener_impl.cpp	2016-04-27 22:34:51.000000000 +0200
+++ libfreenect2-0.2.0-new/src/frame_listener_impl.cpp	2017-11-29 15:56:31.717085590 +0100
@@ -109,7 +109,7 @@
 #ifdef LIBFREENECT2_THREADING_STDLIB
   libfreenect2::unique_lock l(impl_->mutex_);
 
-  auto predicate = std::bind(&SyncMultiFrameListenerImpl::hasNewFrame, impl_);
+  auto predicate = [this](){ return impl_->hasNewFrame(); };
 
   if(impl_->condition_.wait_for(l, std::chrono::milliseconds(milliseconds), predicate))
   {
